Add ability to generate graphs based on correlations of sequences #25933

Bruno-TT opened the pull request:
fixes #25933

Bruno-TT committed:
initial implementation start

Bruno-TT committed:
initial implementation start

Bruno-TT committed:
base test fix

Bruno-TT committed:
base test fix

Bruno-TT committed:
changed class to generator method + tab completion

Bruno-TT committed:
changed class to generator method + tab completion

Bruno-TT committed:
Fixed doctest + improved docstring

Bruno-TT committed:
Fixed doctest + improved docstring

dimpase committed:
Merge branch 'develop' into u/gh-Bruno-TT/correlation-graph

dimpase commented on src/sage/graphs/generators/basic.py:
```diff
@@ -393,6 +395,36 @@ def CompleteGraph(n):
+    adjacency_matrix = Matrix(boolean_adjacency_matrix.astype(int))
+
+    # call graph constructor
+    return Graph(adjacency_matrix, format="adjacency_matrix", name="Correlation Graph")   
```
check the linter error (trailing space) in this line

dimpase replied:
https://github.com/sagemath/sage/actions/runs/4117765743/jobs/7109476469

yyyyx4 commented:
You can use one of [a few select keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link a pull request to an issue. For example, editing the description to include "fixes #25933" will instruct GitHub to automatically close #25933 when this pull request gets merged.

dimpase commented:
fixes #25933

fchapoton committed:
fix the linter once more

Bruno-TT committed:
Merge branch 'develop' into u/gh-Bruno-TT/correlation-graph

Bruno-TT committed:
Merge branch 'sagemath:develop' into u/gh-Bruno-TT/correlation-graph

dimpase committed:
don't import from sage.all

dimpase committed:
don't import from sage.all

dimpase committed:
fix the tests, remove trailing spaces

dimpase committed:
fix the tests, remove trailing spaces

dimpase committed:
fix the linter once more (#35552)

<!-- Please provide a concise, informative and self-explanatory title.
-->
<!-- Don't put issue numbers in the title. Put it in the Description
below. -->
<!-- For example, instead of "Fixes #12345", use "Add a new method to
multiply two integers" -->

### :books: Description

Fixing the linter workflow

<!-- Describe your changes here in detail. -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If this PR resolves an open issue, please link to it here. For
example "Fixes #12345". -->
<!-- If your change requires a documentation PR, please link it
appropriately. -->

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. It should be `[x]` not `[x
]`. -->

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on
- #12345: short description why this is a dependency
- #34567: ...
-->

<!-- If you're unsure about any of these, don't hesitate to ask. We're
here to help! -->

Bruno-TT committed:
Merge branch 'u/gh-Bruno-TT/correlation-graph' of github.com:Bruno-TT/sage into HEAD

Bruno-TT committed:
improved test coverage
